kore/src/GraphTraversal.hs

```diff
@@ -0,0 +1,295 @@
+                    -- Other states may be unfinished but not stuck (Stopped)
+                    -- Only provide the requested amount of states (maxCounterExamples)
+                    unproven = take maxStuck . map (fmap snd) $ filter isStopped results
+                 in if
+                            | (not $ null stuck) -> GotStuck 0 stuck
+                            | not $ null unproven -> GotStuck 0 unproven
+                            | otherwise -> fmap snd result
+            other -> fmap snd other
+
+data StepResult a
+    = Continue (Seq a)
+    | Output (TransitionResult a) (Seq a)
+    | Abort (TransitionResult a) (Seq a)
+    deriving stock (Eq, Show, GHC.Generic)
+
+data TraversalResult a
+    = -- | remaining queue length and stuck or stopped (unproven)
+      -- states (always at most maxCounterExamples many)
+      GotStuck Int [TransitionResult a]
+    | -- | queue length (exceeding the limit) and result(s) of the
+      -- last step that led to stopping
+      Aborted Int [TransitionResult a]
+    | -- | queue empty, results returned
+      Ended [TransitionResult a]
+    deriving stock (Eq, Show, GHC.Generic)
+
+instance Pretty a => Pretty (TraversalResult a) where
+    pretty = \case
+        GotStuck n as ->
+            Pretty.hang 4 . Pretty.vsep $
+                ("Got stuck with queue of " <> Pretty.pretty n) :
+                map Pretty.pretty as
+        Aborted n as ->
+            Pretty.hang 4 . Pretty.vsep $
+                ("Aborted with queue of " <> Pretty.pretty n) :
+                map Pretty.pretty as
+        Ended as ->
+            Pretty.hang 4 . Pretty.vsep $
+                "Ended" : map Pretty.pretty as
+
+instance Functor TraversalResult where
+    fmap f = \case
+        GotStuck n rs -> GotStuck n (ffmap f rs)
+        Aborted n rs -> Aborted n (ffmap f rs)
+        Ended rs -> Ended (ffmap f rs)
+      where
+        ffmap = map . fmap
+
+----------------------------------------
+-- constructing transition functions (for caller)
+
+simpleTransition ::
+    forall config m rule.
+    Monad m =>
+    -- | primitive strategy rule
+    (Prim -> config -> TransitionT rule m config) ->
+    -- | converter to interpret the config (claim state or program state)
+    -- TODO should also consider the applied rule(s) (for Terminal/Cut)
+    (config -> [config] -> TransitionResult config) ->
+    -- final transition function
+    ([Step], config) ->
+    m (TransitionResult ([Step], config))
+simpleTransition applyPrim mapToResult = uncurry tt
+  where
+    tt :: [Step] -> config -> m (TransitionResult ([Step], config))
+    tt [] config =
+        pure $ fmap ([],) $ mapToResult config []
+    tt ([] : iss) config =
+        tt iss config
+    tt (is : iss) config =
+        (fmap (iss,) . mapToResult config . map fst)
+            <$> runTransitionT (applyGroup is config)
+
+    applyGroup :: Step -> config -> TransitionT rule m config
+    applyGroup is c = foldM (flip applyPrim) c is
```

Comment:
This seems to be a problem with `cabal` but not with `stack` - we should consolidate dependency versions more thoroughly IMHO